api-rights-bridge/src/test/java/org/xwiki/contrib/rights/internal/DefaultRightsWriterTest.java

```diff
@@ -409,6 +410,42 @@ void testSaveRuleWithNullRights() throws XWikiException
         assertEquals(1, rightsObj.getIntValue(ALLOW_PROPERTY));
     }
 
+    @Test
+    void testSaveRuleWithEmptyRightsList() throws XWikiException
+    {
+        DocumentReference documentReference = new DocumentReference("xwiki", "S", "P");
+        // prepare rules to put on the document, with empty rights set
+        WritableSecurityRule rule = new WritableSecurityRuleImpl(
+            Collections.singletonList(new DocumentReference("xwiki", "XWiki", "XWikiAdminGroup")),
+            Collections.emptyList(), new RightSet(Right.VIEW), RuleState.ALLOW);
+
+        // call the function under test
+        this.rightsWriter.saveRules(Arrays.asList(rule), documentReference);
+
+        // get the document that was just modified and assert on it
+        XWikiDocument resultDoc =
+            this.oldcore.getSpyXWiki().getDocument(documentReference, this.oldcore.getXWikiContext());
+        // check that there is a single object set
+        assertEquals(1, resultDoc.getXObjects(DefaultRightsWriter.XWIKI_RIGHTS_CLASS).size());
+        BaseObject rightsObj = resultDoc.getXObject(DefaultRightsWriter.XWIKI_RIGHTS_CLASS);
+        // check that the allow is set to default true
+        assertEquals(1, rightsObj.getIntValue(ALLOW_PROPERTY));
+        // and the rest of the object is set properly, as we asked for
+        assertEquals("", rightsObj.getLargeStringValue(USERS_PROPERTY));
+        assertEquals("XWiki.XWikiAdminGroup", rightsObj.getLargeStringValue(GROUPS_PROPERTY));
+        assertEquals(1, rightsObj.getIntValue(ALLOW_PROPERTY));
```

Comment:
We don't need these assertions on the data in the object here: checking that the rule we give to the write API is correctly transformed into corresponding objects is not what is under test here. What is under test here is that following a call with empty list, the rule (whichever that is), is removed. You can keep the assertion that an object is present as we expect it to dissapear later, so it makes sense to make sure that it was ever there...
While I understand the logic of "it doesn't harm to test twice", the harm is actually the fact that the test becomes uselessly complex, harder to read and maintain.